pkg/kubelet/dockershim/docker_container.go

```diff
@@ -0,0 +1,196 @@
+			WorkingDir: config.GetWorkingDir(),
+			Labels:     labels,
+			// Interactive containers:
+			OpenStdin: config.GetStdin(),
+			StdinOnce: config.GetStdinOnce(),
+			Tty:       config.GetTty(),
+		},
+	}
+
+	// Fill the HostConfig.
+	hc := &dockercontainer.HostConfig{
+		Binds:          generateMountBindings(config.GetMounts()),
+		ReadonlyRootfs: config.GetReadonlyRootfs(),
+		Privileged:     config.GetPrivileged(),
+	}
+
+	// Apply options derived from the sandbox config.
+	if lc := sandboxConfig.GetLinux(); lc != nil {
+		// Apply Cgroup options.
+		// TODO: Check if this works with per-pod cgroups.
+		hc.CgroupParent = lc.GetCgroupParent()
+
+		// Apply namespace options.
+		sandboxNSMode := fmt.Sprintf("container:%v", podSandBoxID)
+		hc.NetworkMode = dockercontainer.NetworkMode(sandboxNSMode)
+		hc.IpcMode = dockercontainer.IpcMode(sandboxNSMode)
+		hc.UTSMode = ""
+		hc.PidMode = ""
+
+		nsOpts := lc.GetNamespaceOptions()
+		if nsOpts != nil {
+			if nsOpts.GetHostNetwork() {
+				hc.UTSMode = namespaceModeHost
+			}
+			if nsOpts.GetHostPid() {
+				hc.PidMode = namespaceModeHost
+			}
+		}
+	}
+
+	// Apply Linux-specific options if applicable.
+	if lc := config.GetLinux(); lc != nil {
+		// Apply resource options.
+		// TODO: Check if the units are correct.
+		// TODO: Can we assume the defaults are sane?
+		rOpts := lc.GetResources()
+		if rOpts != nil {
+			hc.Resources = dockercontainer.Resources{
+				Memory:     rOpts.GetMemoryLimitInBytes(),
+				MemorySwap: -1, // Always disable memory swap.
+				CPUShares:  rOpts.GetCpuShares(),
+				CPUQuota:   rOpts.GetCpuQuota(),
+				CPUPeriod:  rOpts.GetCpuPeriod(),
+				// TODO: Need to set devices.
+			}
+			hc.OomScoreAdj = int(rOpts.GetOomScoreAdj())
+		}
+		// Note: ShmSize is handled in kube_docker_client.go
+	}
+
+	// TODO: Seccomp support. Need to figure out how to pass seccomp options
+	// through the runtime API (annotations?).See dockerManager.getSecurityOpts()
+	// for the details. Always set the default seccomp profile for now.
+	hc.SecurityOpt = []string{fmt.Sprintf("%s=%s", "seccomp", defaultSeccompProfile)}
+
+	createConfig.HostConfig = hc
+	createResp, err := ds.client.CreateContainer(createConfig)
+	if createResp != nil {
+		return createResp.ID, err
+	}
+	return "", err
+}
+
+// StartContainer starts the container.
+func (ds *dockerService) StartContainer(rawContainerID string) error {
+	return ds.client.StartContainer(rawContainerID)
+}
+
+// StopContainer stops a running container with a grace period (i.e., timeout).
+func (ds *dockerService) StopContainer(rawContainerID string, timeout int64) error {
+	return ds.client.StopContainer(rawContainerID, int(timeout))
+}
+
+// RemoveContainer removes the container. If the container is running, the container
+// should be force removed.
+func (ds *dockerService) RemoveContainer(rawContainerID string) error {
+	return ds.client.RemoveContainer(rawContainerID, dockertypes.ContainerRemoveOptions{RemoveVolumes: true})
```

Comment:
Hm....I don't think we need that. I copied the comment from @feiskyer's PR. I'll add a new comment.